src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -266,6 +268,32 @@ public bool IsNullable
         }
 
         /// <summary>
+        /// Gets a value indicating whether this is a generic definition, or
+        /// an instance of System.ByReference`1.
+        /// </summary>
+        public bool IsByReferenceOfT
+        {
+            get
+            {
+                return this.GetTypeDefinition().IsWellKnownType(WellKnownType.ByReferenceOfT);
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this is a byref-like type (a <code>TypedReference</code>, <code>Span&lt;T&gt;</code>, etc.).
+        /// </summary>
+        public bool IsByRefLike
+        {
+            get
+            {
+                // TODO: should we key this off of the fact that there's an instance field
```

Comment:
Yes it should. ByRefLike should be set to true for Span<T>, ByReference<T>, any generic structure that has a Span<T> or ByReference<T> in it.